src/core/trace_core.ml

```diff
@@ -55,6 +55,16 @@ let[@inline] exit_manual_span espan : unit =
   | None -> ()
   | Some (module C) -> C.exit_manual_span espan
 
+let[@inline] add_data_to_span (span : span) data : unit =
+  if data <> [] then (
+    match A.get collector with
+    | None -> ()
+    | Some (module C) -> C.add_data_to_span span data
+  )
```

Comment:
Hm. In an implicit (non-`explicit_span`) context, making sure the metadata is added to the _specifically requested_ span might actually be a … little more involved. I think I can pull it off; but I was actually assuming we'd have two APIs:

1. "add data to the current implicit span" (`data -> unit`);
2. "add data to <given> explicit span (`explicit_span -> data -> unit`)

But the middle-ground, `span -> data -> unit`, feels a little stranger to me? I think it's workable, it's just definitely disconnected from the implementation details I'll have to write, and feels slightly out-of-line with the fact that `with_span` is currently entirely implicit …

(This makes me even more sure that `Trace.Implicit` and `Trace.Explicit` are a good idea!)